Add a way of creating and executing "config switch" operations

barthisrael opened the pull request:
Through this PR we expose an operation in the REST API for performing `barman config-switch` command remotely. The following changes have been applied:

* Create a ``ConfigSwitchOperation`` class, which takes care of validating the arguments for a config switch operation, and also of running such operation;
* Modify `servers_operations_post` Flask end point so it is able to receive POST requests both for creating a recovery operation or a config switch operation;
* Create a new command ``pg-backup-api config-switch``, which is used by the API in order to run the config switch on Barman;
* Create a function ``config_switch_operation``, which is used by ``pg-backup-api config-switch`` command;
* Create a helper function ``_run_operation``, which is used both by ``recovery_operation`` and ``config_switch_operation`` functions. They both have a very similar logic, so that helper function executes the common code paths.

The new API endpoint is able to handle both an "apply model" or a "reset model" through `barman config-switch` command.

References: BAR-125.

gcalacoci reviewed (approved):
this looks good to me for now

barthisrael commented:
> this looks good to me for now

Thanks for checking :)

Once https://github.com/EnterpriseDB/pg-backup-api/pull/87 is merged, I'll rebase this branch on top of that.

barthisrael commented:
Rebased on top of `main` to include changes from #87 . Based on that:

* Adjusted unit tests of `test_server_operation.py` to use `pytest` and split complex tests
* Included missing unit tests for `test_main.py`, `test_utility_controller.py`, and `test_run.py` -- modules which were introduced by #87 .

I plan to rebase this branch on top of `main` again in the future, once #88 is merged into `main`, because that PR also touches some unit tests and the changes are relevant here (in regards to patching terminal size).

barthisrael commented:
Rebased on top of `main` after #88 has been merged, and adjusted the code/unit tests accordingly.
